orm/src/main/java/org/hibernate/search/jmx/impl/IndexControl.java

```diff
@@ -95,33 +96,27 @@ public void index(String entity) {
 			Thread.currentThread().interrupt();
 			throw new SearchException( "Unable to complete indexing" );
 		}
-		session.close();
 	}
 
 	@Override
 	public void optimize(String entity) {
 		Class<?> clazz = getEntityClass( entity );
-
 		SessionFactory factory = getSessionFactory();
-		Session session = factory.openSession();
-		FullTextSession fullTextSession = Search.getFullTextSession( session );
-		fullTextSession.beginTransaction();
-		fullTextSession.getSearchFactory().optimize( clazz );
-		fullTextSession.getTransaction().commit();
-		session.close();
+		ExtendedSearchIntegrator searchIntegrator = ContextHelper.getSearchIntegratorBySF( factory );
+		searchIntegrator.optimize( clazz );
 	}
 
 	@Override
 	public void purge(String entity) {
 		Class<?> clazz = getEntityClass( entity );
 
 		SessionFactory factory = getSessionFactory();
-		Session session = factory.openSession();
-		FullTextSession fullTextSession = Search.getFullTextSession( session );
-		fullTextSession.beginTransaction();
-		fullTextSession.purgeAll( clazz );
-		fullTextSession.getTransaction().commit();
-		session.close();
+		try ( Session session = factory.openSession() ) {
```

Comment:
@Sanne Interestingly enough, in the lastest versions of ORM, this seems to be fixed:

 * `SessionImpl` calls `super.close()`: https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java#L400
 * The superclass' `close()` method simply returns if the session is already closed: https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/internal/AbstractSharedSessionContract.java#L289

The issue is still there in the ORM version we're using, though.